### PR TITLE
[15.0][FIX] hr_holidays_natural_period: Avoid error when saving leave

### PR DESCRIPTION
**Pull request by victoralmau**
Avoid error when saving leave.

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT42608

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]
Thanks :)

File: hr_holidays_natural_period/models/hr_leave_type.py [hunk 1]
```diff
@@ -11,3 +11,23 @@ class HrLeaveType(models.Model):
...
+    def _get_employees_days_per_allocation(self, employee_ids, date=None):
+        """We need to set request_unit as 'day' to avoid the calculations being done
+        as hours.
+        Related code:
+        hr_holidays/models/hr_leave_type.py#L313
+        hr_holidays/models/hr_leave_type.py#L367
+        """
+        old_request_unit_data = {}
+        for item in self:
+            old_request_unit_data[item.id] = item.request_unit
```
Comment (pedrobaeza) [inline hr_holidays_natural_period/models/hr_leave_type.py:24]: Don't store (and change) all the types, just those with natural days.

File: hr_holidays_natural_period/__manifest__.py [hunk 2]
```diff
@@ -12,5 +12,5 @@
     "installable": True,
     "depends": ["hr_holidays"],
     "maintainers": ["victoralmau"],
-    "demo": ["demo/hr_leave_type_data.xml"],
+    "demo": ["demo/hr_leave_type_demo.xml"],
```
Comment (pedrobaeza) [inline hr_holidays_natural_period/__manifest__.py]: This change requires a migration script to try to delete the type.

Comment (victoralmau) [reply hr_holidays_natural_period/__manifest__.py]: I have left the file name as it was.

Comment (pedrobaeza) [reply hr_holidays_natural_period/__manifest__.py]: No, it's correct to be demo data, as it was an error to put it as data, but it should delete old occurrences through delete_safely... method in openupgradelib.

Comment (victoralmau) [reply hr_holidays_natural_period/__manifest__.py]: I'm sorry but I don't understand.

They have always been defined as demo data, now we are just renaming the file to make it more consistent.

Comment (pedrobaeza) [reply hr_holidays_natural_period/__manifest__.py]: Ah, OK, my fault. Sorry, I saw data > demo, and I think this is changing the other part (the key, changing its meaning). It's OK then to just rename and nothing more.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge patch

**Pull request merged**